### PR TITLE
make failing test showing spurious variable conflicts

### PR DESCRIPTION
**Pull request by tkellen**
Per @monopole's request in https://github.com/kubernetes-sigs/kustomize/pull/1620/files#r334282443, this PR isolates a failing test for gh-1600.

### CODE REVIEW

**Comment by tkellen** [comment]
Not sure I went as far as you'd like with the functional "sugar" for the tests @monopole, but I tried to clean it up in a way that I felt would be much more readable for future maintainers.

File: pkg/accumulator/resaccumulator_test.go [hunk 1]
```diff
@@ -179,6 +179,86 @@ func TestResolveVarsVarNeedsDisambiguation(t *testing.T) {
...
+		},
+		"data": map[string]interface{}{
+			"provider": value,
+		},
+	}
+}
+
+func makeVarToNamepaceAndPath(
+	name string,
+	namespace string,
```
Comment (monopole) [inline pkg/accumulator/resaccumulator_test.go]: namespace isn't used, did you mean env?

File: pkg/accumulator/resaccumulator_test.go [hunk 2]
```diff
@@ -179,6 +179,86 @@ func TestResolveVarsVarNeedsDisambiguation(t *testing.T) {
...
+		Name: name,
+		ObjRef: types.Target{
+			Gvk:  gvk.Gvk{Version: "v1", Kind: "ConfigMap"},
+			Name: "environment",
+		},
+		FieldRef: types.FieldSelector{FieldPath: path},
+	}
+}
+
+func TestResolveVarConflicts(t *testing.T) {
```
Comment (monopole) [inline pkg/accumulator/resaccumulator_test.go]: this test reads like an example, which is perfect.  thanks!

File: pkg/accumulator/resaccumulator_test.go (hunk 1 above)
Comment (tkellen) [reply pkg/accumulator/resaccumulator_test.go]: Whoops. No, I did mean to use it, though clearly it isn't strictly required I felt it helped illustrate my particular incarnation of this issue. I've updated this.

File: pkg/accumulator/resaccumulator_test.go (hunk 2 above)
Comment (tkellen) [reply pkg/accumulator/resaccumulator_test.go]: My pleasure, thanks for prompting me to take a bit more care.

File: pkg/accumulator/resaccumulator_test.go [hunk 3]
```diff
@@ -179,6 +179,87 @@ func TestResolveVarsVarNeedsDisambiguation(t *testing.T) {
...
+	rm1.Append(rf.FromMap(barAws))
+	ac1 := MakeEmptyAccumulator()
+	ac1.AppendAll(rm1)
+	ac1.MergeVars([]types.Var{varBar})
+
+	// validate that two vars of the same name which reference the same concrete
+	// value do not produce a conflict.
+	err := ac0.MergeAccumulator(ac1)
+	if err != nil {
+		t.Fatalf("dupe var names w/ same concrete val should not conflict: %v", err)
```
Comment (monopole) [inline pkg/accumulator/resaccumulator_test.go]: so the current form of this test should be
```
if err == nil {
  t.Fatalf("see bug whatever")
}
```

the subsequent PR to fix that bug should change this line.

we cannot check in failing tests, but we can check in tests that succeed showing behavior we want to change

Comment (tkellen) [reply pkg/accumulator/resaccumulator_test.go]: Got it. Updated this and the now-dependent PR.

**Pull request merged**